src/pyotp/totp.py

```diff
@@ -32,6 +32,11 @@ def __init__(
         """
         if digest is None:
             digest = hashlib.sha1
+        elif digest in [
+            hashlib.md5,
+            hashlib.shake_128
+        ]:
+            raise ValueError("selected digest function must generate digest size greater than or equals to 18 bytes")
```

Comment:
I think this can go away if the check is one in `otp.py:generate_otp()`: It is not necessary, and there should only be one place where the relevant check is done.